sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -66,7 +66,13 @@ case class WindowSpecDefinition(
   override def checkInputDataTypes(): TypeCheckResult = {
     frameSpecification match {
       case UnspecifiedFrame =>
-        DataTypeMismatch(errorSubClass = "UNSPECIFIED_FRAME")
+        DataTypeMismatch(
+          errorSubClass = "INTERNAL_ERROR",
+          messageParameters = Map(
+            "message" -> ("Cannot use an UnspecifiedFrame. " +
+              "This should have been converted during analysis.")
```

Comment:
cannot throw an exception here, otherwise cannot continue

Reply:
https://github.com/apache/spark/blob/0eaa8e1e76ab6ecdd3b51d751857e50530ccdeb6/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala#L59-L61

Reply:
Not sure this is a good idea ... cc @MaxGekk @itholic

Reply:
Both checks exist in the source code from the first commit:
https://github.com/apache/spark/commit/f2c47082c3412a4cf8cbabe12585147c5ec3ea40
I would leaved `case UnspecifiedFrame` as is to cover the bad situation explicitly.

Reply:
boolean short circuit

Reply:
It is more reasonable to check `frameSpecification.isInstanceOf[SpecifiedWindowFrame]` before `checkInputDataTypes()`